test/Libraries/Microsoft.Extensions.AI.Tests/ChatCompletion/ChatClientStructuredOutputExtensionsTests.cs

```diff
@@ -172,6 +172,40 @@ public async Task CanUseNativeStructuredOutput()
         Assert.Equal("Hello", Assert.Single(chatHistory).Text);
     }
 
+    [Fact]
+    public async Task CanUseNativeStructuredOutputWithSanitizedTypeName()
+    {
+        var expectedResult = new Data<Animal> { Value = new Animal { Id = 1, FullName = "Tigger", Species = Species.Tiger } };
+        var expectedCompletion = new ChatCompletion([new ChatMessage(ChatRole.Assistant, JsonSerializer.Serialize(expectedResult))]);
+
+        using var client = new TestChatClient
+        {
+            CompleteAsyncCallback = (messages, options, cancellationToken) =>
+            {
+                var responseFormat = Assert.IsType<ChatResponseFormatJson>(options!.ResponseFormat);
+
+                Assert.Matches("^[a-zA-Z0-9_-]+$", responseFormat.SchemaName);
```

Comment:
Nit: could you assert the actual name being returned?

Reply:
Sure, but I thought it was better to make the test more resilient to .net metadata naming since the point is actually the regex sanitization. Perhaps a theory would make that more obvious? 

But I'll change it if you think that is better, sure.

Reply:
I think it would be valuable to verify that the output of the sanitization is something expected rather than any string that happens to be validated by the regex. I don't mind using a theory if you would like to test this against more inputs.

Reply:
Done!